site/docs/relations/physical_relations.md

```diff
@@ -17,13 +17,15 @@ The hash equijoin join operator will build a hash table out of the right input b
 
 ### Hash Equijoin Properties
 
-| Property            | Description                                                  | Required                 |
-| ------------------- | ------------------------------------------------------------ | ------------------------ |
-| Left Input          | A relational input.                                          | Required                 |
-| Right Input         | A relational input.                                          | Required                 |
-| Join Expression     | A boolean condition that describes whether each record from the left set "match" the record from the right set. The condition must only include the following operations: AND, ==, field references, is not distinct from. Field references correspond to the direct output order of the data. | Required.                |
-| Post Join Predicate | An additional expression that can be used to reduce the output of the join operation post the equality condition. Minimizes the overhead of secondary join conditions that cannot be evaluated using the equijoin keys. | Optional, defaults true. |
-| Join Type           | One of the join types defined in the Join operator.          | Required                 |
+| Property            | Description                                                                                                                                                                                                                                                                                     | Required                                                                   |
+|---------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------|
+| Left Input          | A relational input.                                                                                                                                                                                                                                                                             | Required                                                                   |
+| Right Input         | A relational input.                                                                                                                                                                                                                                                                             | Required                                                                   |
```

Comment:
Since this is a physical relation do we want to explicitly declare that the left side is the build side and the right side is the probe side (or vice versa)?  It looks like you are already making this assumption in your description of left keys and right keys.

Reply:
I think we can leave it to the user to decide which side is probe or build , your thoughts?

Reply:
Picking the correct side is generally an optimizer task (requires statistics / heuristics).  My assumption for more physical nodes is that they've already gone through an optimizer.

For a single optimizer/consumer pair this wouldn't be a problem.  They can both agree offline which side should be build and probe.  However, imagine an engine-independent substrait optimizer created this plan and hoped for it to be consumed by many different engines.  In that case the different consumers might pick left/right build/probe differently.

That being said, I haven't worked much with optimizers.  CC @jacques-n who has more experience here.